src/compiler/types.ts

```diff
@@ -1594,6 +1664,8 @@ module ts {
         assignmentChecks?: Map<boolean>;  // Cache of assignment checks
         hasReportedStatementInAmbientContext?: boolean;  // Cache boolean if we report statements in ambient context
         importOnRightSide?: Symbol;       // for import declarations - import that appear on the right side
+        jsxFlags?: JsxFlags;              // flags for knowning what kind of element/attributes we're dealing with

```

Comment:
You could do this, or you could mix the JsxFlags into the NodeCheckFlags and store it on the same `flags` property. Personally I actually prefer what you did, just saying.